packages/date-picker/test/validation.test.js

```diff
@@ -116,16 +118,14 @@ describe('validation', () => {
       expect(validateSpy.called).to.be.false;
     });
 
-    it('should validate on clear button click', async () => {
+    it('should validate before change event on clear button click', async () => {
       datePicker.clearButtonVisible = true;
-      // Set invalid value.
-      setInputValue(datePicker, 'foo');
```

Comment:
The clear button is only visible when date-picker has a valid value, so what this test does isn't a valid use case. However, since there wasn't any test ensuring that the clear button clears the input element's value, I added a respective assertion to:

- `should clear the value on click`
- `should clear the value on touch tap`